test_mixpanel.py

```diff
@@ -26,11 +28,13 @@ def send(self, endpoint, event, api_key=None):
 
 # Convert a query string with base64 data into a dict for safe comparison.
 def qs(s):
+    if isinstance(s, six.binary_type):
+        s = s.decode('utf8')
     blob = cgi.parse_qs(s)
-    if 'data' in blob:
-        if len(blob['data']) != 1:
-            pytest.fail('found multi-item data: %s' % blob['data'])
-        blob['data'] = json.loads(base64.b64decode(blob['data'][0]))
+    if len(blob['data']) != 1:
+        pytest.fail('found multi-item data: %s' % blob['data'])
+    json_bytes = base64.b64decode(blob['data'][0])
+    blob['data'] = json.loads(json_bytes.decode('utf8'))
```

Comment:
@joeatwork Split into two lines here, which hopefully makes it at least slightly clearer. There's a lot of en/de-coding to do!